frontend/src/scenes/dashboard/dashboardLogic.ts

```diff
@@ -185,12 +180,17 @@ export const dashboardLogic = kea<dashboardLogicType<DashboardLogicProps>>({
                         items: state?.items.map((item) => ({ ...item, layouts: itemLayouts[item.short_id] })),
                     } as DashboardType
                 },
-                [dashboardsModel.actionTypes.updateDashboardItem]: (state, { item }) => {
+                [dashboardsModel.actionTypes.updateDashboardItem]: (state, { item, dashboardIds }) => {
+                    if (dashboardIds && props.id && !dashboardIds.includes(props.id)) {
+                        // this update is not for this dashboard
+                        return state
+                    }
+
                     if (state) {
                         const itemIndex = state.items.findIndex((i) => i.short_id === item.short_id)
                         const newItems = state.items.slice(0)
                         if (itemIndex >= 0) {
-                            newItems[itemIndex] = item
+                            newItems[itemIndex] = withDataFromDashboardContext<InsightModel>(item, newItems[itemIndex])
```

Comment:
If your function uses type generics like `T`, you don't really need to specify `<InsightModel>` here, it should get autodetected...?